std/stdio.d

```diff
@@ -843,12 +844,12 @@ Throws: $(D ErrnoException) on failure if closing the file.
     void detach() @safe
     {
         if (!_p) return;
-        if (_p.refs == 1)
+        if (atomicLoad(_p.refs) == 1)
```

Comment:
This is a potential resource leak. Example:

Two threads have references. Both check count, and see that it is 2. Both skip this branch and so the file is not closed. What needs to instead happen is unconditional `atomic!"-="` and check the result of that.

So in the new example:
Two threads have references. Both atomically decrement. One sees `0` as the new value and the other sees `1` as the new value. Whichever sees `0` then calls close.